sdk/core/azure-core/azure/core/messaging.py

```diff
@@ -122,6 +122,19 @@ def from_dict(cls, event):
         :type event: dict
         :rtype: CloudEvent
         """
+        # https://github.com/cloudevents/spec Cloud event spec requires source, type,
+        # specversion. We autopopulate everything other than source, type.
+        if not all([_ in event for _ in ("source", "type")]):
+            if all([_ in event for _ in ("subject", "eventType", "data", "dataVersion", "id", "eventTime")]):
+                raise ValueError(
+                    "The event does not conform to the cloud event spec https://github.com/cloudevents/spec." +
+                    " Try using the EventGridEvent from azure-eventgrid library"
+                )
+            raise ValueError(
```

Comment:
This is client side validation. I would do this whole block into a try/except around the `cls` at the end